src/com/company/ParseTest.java

```diff
@@ -0,0 +1,106 @@
+package com.company;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ParseTest {
+
+    @Test
+    public void doesNotExist() throws Exception {
```

Comment:
It might be useful to separate the "doesn't exist" case from the "malformed code" case. On line 15, that's an example of doesn't exist, while lines 16-17 likely won't compile or possibly even won't parse.

It would also be useful to check a case where the variable name isn't even present in the code at all.